### ISSUE TITLE
Getting error "Cannot open include file: 'basetsd.h': No such file or directory" when compiling LAVFilters

### ISSUE THREAD

**Issue by kenny1983** [clsid2/mpc-hc#829]
I have spent **an entire day** trying to get this project to compile. Half of the reason for that is probably that I'm using a complex root directory instead of the suggested `C:\mpc-hc`,  but the rest I blame solely on stupid VS 2019 bugs in recent updates, and silly oversights made by either the devs of this project, or the devs of its submodules' projects.

The former was solved by installing VS 2017 alongside VS 2019 using Visual Studio Installer. I selected all of the dependencies mentioned in `Compilation.md` again for this installation, so perhaps I now have two instances of the SDKs which are causing VS 2017 to get confused.

I have posted a comment on https://github.com/clsid2/mpc-hc/issues/328 re: an instance of the latter. This needs to be fixed ASAP IMO!

Anyway, the (hopefully) last error that I'm encountering is (long root path removed for brevity):
```
Severity  Code   Description                                                                                                                   Project     File					 		Line
Error     C1083  Cannot open include file: 'basetsd.h': No such file or directory [src\thirdparty\LAVFilters\src\libbluray\libbluray.vcxproj]  LAVFilters  src\thirdparty\LAVFilters\src\libbluray\config.h     157
```
This error appears a total of **52 times**, which just goes to show how crucial that one header file is, and how insane it is that I'm getting these errors, **yet no one else is**.

If I open the `LAVFilters.sln` solution directly and try to build that instead, I get:
```
Severity  Code   Description                                                                  Project		File							 Line
Error     C1083  Cannot open include file: 'libavutil/avconfig.h': No such file or directory  DSUtilLite	src\thirdparty\lavfilters\src\ffmpeg\libavutil\common.h	 45
```
And that one should be a little easier to solve, since it only appears 6 times across the `DSUtilLite`, `LAVAudio`, `LAVVideo`, `Demuxers` and `LAVSplitter` projects.

I will continue to debug these issues on my own, since I've come this far and am well-and-truly-determined to get this evil beast of a project to compile. But in the meantime, any shoves in the right direction would be very much appreciated. MTIA 😄

**Comment by clsid2** [comment]
You don't need to compile LAV Filters. You can successfully compile MPC-HC without it.

The first error might be due to an issue with the libbluray submodule.
The second error is because you need to build FFmpeg libraries first before LAV Filters. That is done by `build_lavfilters.bat`

Are you planning on contributing patches to MPC-HC? Because I don't have time to babysit people who are just trying to compile "for fun". Sorry for sounding like an asshole.

**Comment by kenny1983** [comment]
LOL yes I wouldn't have come here bitching about this stuff unless I was trying to contribute. And I'm sure you can tell from the tone of my comments that I do not consider the experience so far to have been "for fun" 😉

Thanks for the pointers, but I'm not going to go into why they aren't going to help me...it seems as though that would fall upon deaf ears.

I will grant you this though: it is clearly stated all over the issues and the docs that LAVFilters isn't really needed, and I guess I'm just being stubborn. I'm the kind of guy who, when something doesn't work, wants to know **why** it doesn't work, so that I may be able to fix it, instead of letting it go and letting some other poor shlub deal with it in the future.

**Comment by clsid2** [comment]
LAV Filters is an optional external component. If you want to contribute changes to that component, then that is best done directly in its original project and not our slightly modified clone.

I am using VS2019 16.6.4. I have uploaded its installer for you [LINK](https://www.sendspace.com/file/yaac71).

**Comment by clsid2** [comment]
The basetsd.h error is because VS can't find the Windows SDK.

**Comment by kenny1983** [comment]
> LAV Filters is an optional external component. If you want to contribute changes to that component, then that is best done directly in its original project and not our slightly modified clone.

Duly noted 😄 

> I am using VS2019 16.6.4. I have uploaded its installer for you [LINK](https://www.sendspace.com/file/yaac71).

Thanks, I struggled to find an installer for an older 2019 build, hence why I'm using 2017. And while I appreciate the effort that you went to to provide that installer for me, it doesn't actually work, because it seems MS is now forcing everyone to be on 16.8.3 (or whatever the latest release of a major version is) through Visual Studio Installer 😢.

> The basetsd.h error is because VS can't find the Windows SDK.

Yeah I worked that out by looking at the project properties, but can't understand for the life of me *why* VS can't find it...if I open up `libbluray.vcxproj` in a separate instance and head to Project Properties -> Configuration Properties -> VC++ Directories -> Include Directories, the `$(WindowsSDK_IncludePath)` variable included there resolves to (among many others) the correct 8.1 SDK path (`C:\Program Files (x86)\Windows Kits\8.1\Include\shared`, which definitely does contain the header). That's what got me so confused in the first place and made me want to create this issue.

Anyway I apologise for having wasted your time, and thank you again for your help. I'll just build in Debug_Lite mode for now and come back to this later 😉
